support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/web/flow/DelegatedClientAuthenticationAction.java

```diff
@@ -145,8 +149,13 @@ protected Event doExecute(final RequestContext context) throws Exception {
                 final ClientCredential clientCredential = new ClientCredential(credentials);
                 final AuthenticationResult authenticationResult =
                         this.authenticationSystemSupport.handleAndFinalizeSingleAuthenticationTransaction(service, clientCredential);
-                final TicketGrantingTicket tgt = this.centralAuthenticationService.createTicketGrantingTicket(authenticationResult);
-                WebUtils.putTicketGrantingTicketInScopes(context, tgt);
+                try {
+                    final TicketGrantingTicket tgt = this.centralAuthenticationService.createTicketGrantingTicket(authenticationResult);
+                    WebUtils.putTicketGrantingTicketInScopes(context, tgt);
+                } catch (final PrincipalException e) {
```

Comment:
This probably should be catching `Exception`. Right?

Reply:
I catch only the exception I need in that case and I redirect it to the proper page, if I catch more, I am not sure the page will represent really the exception. As you wish, tell me and I'll make it.

Reply:
Cool. Thanks for the note. This is fine as is.